"warning: compound assignment with 'volatile'-qualified left operand is deprecated" when building with C++20

**Describe the bug**
Building with C++20 issues a warning

**To Reproduce**
build some C++ code with C++20 that includes zephyr.h

**Expected behavior**
Build without warnings.

**Impact**

**Logs and console output**
```
modules/hal/cmsis/CMSIS/Core/Include/mpu_armv7.h:208:14: warning: compound assignment with 'volatile'-qualified left operand is deprecated [-Wvolatile]
  208 |   SCB->SHCSR &= ~SCB_SHCSR_MEMFAULTENA_Msk;
```

--- a/cmake/compiler/gcc/compiler_flags.cmake
+++ b/cmake/compiler/gcc/compiler_flags.cmake
@@ -124,7 +124,8 @@ set_property(TARGET compiler-cpp PROPERTY dialect_cpp98 "-std=c++98")
 set_property(TARGET compiler-cpp PROPERTY dialect_cpp11 "-std=c++11" "-Wno-register")
 set_property(TARGET compiler-cpp PROPERTY dialect_cpp14 "-std=c++14" "-Wno-register")
 set_property(TARGET compiler-cpp PROPERTY dialect_cpp17 "-std=c++17" "-Wno-register")
-set_property(TARGET compiler-cpp PROPERTY dialect_cpp2a "-std=c++2a" "-Wno-register")
+set_property(TARGET compiler-cpp PROPERTY dialect_cpp2a "-std=c++2a"
+  "-Wno-register" "-Wno-volatile")
 
 # Disable exeptions flag in C++
 set_property(TARGET compiler-cpp PROPERTY no_exceptions "-fno-exceptions")

